resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -333,7 +340,7 @@ private[spark] class Client(
     val maxMem = newAppResponse.getMaximumResourceCapability().getMemory()
     logInfo("Verifying our application has not requested more than the maximum " +
       s"memory capability of the cluster ($maxMem MB per container)")
-    val executorMem = executorMemory + executorMemoryOverhead
+    val executorMem = executorMemory + executorMemoryOverhead + pysparkWorkerMemory
     if (executorMem > maxMem) {
       throw new IllegalArgumentException(s"Required executor memory ($executorMemory" +
         s"+$executorMemoryOverhead MB) is above the max threshold ($maxMem MB) of this cluster! " +
```

Comment:
Should add `pysparkWorkerMemory` here too.

Reply:
Maybe just switch it to use the total `$executorMem` instead?

Reply:
I like having it broken out so users can see where their allocation is going. Otherwise, users that only know about `spark.executor.memory` might not know how their allocation is 1gb higher when running PySpark. I've updated this to include the worker memory.